test/datasource.test.js

```diff
@@ -352,4 +352,86 @@ describe('DataSource', function() {
         .should.not.containEql('TestModel');
     });
   });
+
+  describe('execute', () => {
+    let ds;
+    beforeEach(() => ds = new DataSource('ds', {connector: 'memory'}));
+
+    it('calls connnector to execute the command', () => {
+      let called = 'not called';
+      ds.connector.execute = function(command, args, options, callback) {
+        called = {command, args, options};
+        callback(null, 'a-result');
+      };
+
+      return ds.execute('command', ['arg1', 'arg2'], {'a-flag': 'a-value'})
+        .then(result => {
+          result.should.be.equal('a-result');
+          called.should.be.eql({
+            command: 'command',
+            args: ['arg1', 'arg2'],
+            options: {'a-flag': 'a-value'},
+          });
+        });
+    });
+
+    it('supports shorthand version (cmd)', () => {
+      let called = 'not called';
+      ds.connector.execute = function(command, args, options, callback) {
+        called = {command, args, options};
+        callback(null, 'a-result');
+      };
+
+      return ds.execute('command')
+        .then(result => {
+          result.should.be.equal('a-result');
+          called.should.be.eql({
+            command: 'command',
+            args: [],
+            options: {},
+          });
+        });
+    });
+
+    it('supports shorthand version (cmd, args)', () => {
+      let called = 'not called';
+      ds.connector.execute = function(command, args, options, callback) {
+        called = {command, args, options};
+        callback(null, 'a-result');
+      };
+
+      return ds.execute('command', ['arg1', 'arg2'])
+        .then(result => {
+          result.should.be.equal('a-result');
+          called.should.be.eql({
+            command: 'command',
+            args: ['arg1', 'arg2'],
+            options: {},
+          });
+        });
+    });
+
+    it('converts multiple callbacks arguments into a promise resolved with an array', () => {
+      ds.connector.execute = function(command, args, options, callback) {
+        callback(null, 'result1', 'result2');
+      };
+      return ds.execute('command').then(result => {
+        result.should.eql(['result1', 'result2']);
+      });
+    });
+
+    it('throws NOT_IMPLEMENTED when no connector is provided', () => {
+      ds.connector.execute = undefined;
+      return ds.execute('command').should.be.rejectedWith({
+        code: 'NOT_IMPLEMENTED',
+      });
+    });
+
+    it('throws NOT_IMPLEMENTED for connectors  not implementing execute', () => {
+      ds.connector = undefined;
```

Comment:
I think the name of this test should switch with the one above :)

- this one tests connector not provided.
- the one above tests execute not implemented.